UtopiaCity/Controllers/Sport/SportComplexController.cs

```diff
@@ -1,62 +1,80 @@
+using System.Collections.Generic;
 using UtopiaCity.Models.Sport;
 using UtopiaCity.Services.Sport;
+using UtopiaCity.ViewModels.Sport;
 
 namespace UtopiaCity.Controllers.Sport
 {
     public class SportComplexController : Controller
     {
         private readonly SportComplexService _sportComplexService;
+        private readonly IMapper _mapper;
 
-        public SportComplexController(SportComplexService sportComplexService)
+        public SportComplexController(SportComplexService sportComplexService, IMapper mapper)
         {
             _sportComplexService = sportComplexService;
+            _mapper = mapper;
         }
 
-        public IActionResult AllSportComplexes() => View(_sportComplexService.GetAllSportComplexes());
+        public IActionResult AllSportComplexes()
+        {
+            List<SportComplex> sportComplexes = _sportComplexService.GetAllSportComplexes();
+            List<SportComplexViewModel> sportComplexViewModels = new List<SportComplexViewModel>();
+            foreach (SportComplex sportComplex in sportComplexes)
+            {
+                sportComplexViewModels.Add(_mapper.Map<SportComplexViewModel>(sportComplex));
+            }
+
+            return View(sportComplexViewModels);
+        }
 
         public IActionResult Details(string id)
         {
-            var sportComplex = _sportComplexService.GetSportComplexById(id);
+            SportComplex sportComplex = _sportComplexService.GetSportComplexById(id);
             if (sportComplex == null)
             {
                 return NotFound();
             }
 
-            return View(sportComplex);
+            SportComplexViewModel sportComplexViewModel = _mapper.Map<SportComplexViewModel>(sportComplex);
+            return View(sportComplexViewModel);
         }
 
         [HttpGet]
         public IActionResult Create() => View();
 
         [HttpPost]
-        public IActionResult Create(SportComplex sportComplex)
+        public IActionResult Create(SportComplexViewModel sportComplexViewModel)
         {
-            if (ModelState.IsValid && sportComplex != null)
+            if (sportComplexViewModel == null)
             {
-                _sportComplexService.AddSportComplexToDb(sportComplex);
-                return RedirectToAction(nameof(AllSportComplexes));
+                return View("Error", "You made mistakes while creating new Sport Complex");
             }
 
-            return View("Error", "You made mistakes while creating new Sport Complex");
+            SportComplex sportComplex = _mapper.Map<SportComplex>(sportComplexViewModel);
+            _sportComplexService.AddSportComplexToDb(sportComplex);
+            return RedirectToAction(nameof(AllSportComplexes));
         }
 
         [HttpGet]
         public IActionResult Delete(string id)
         {
-            var sportComplex = _sportComplexService.GetSportComplexById(id);
+            SportComplex sportComplex = _sportComplexService.GetSportComplexById(id);
```

Comment:
probably it is clear that type is SportComplex, var was appropriate

Reply:
I thought, it will be more clear